go/arrow/flight/flightsql/driver/driver.go

```diff
@@ -79,28 +96,37 @@ func (r *Rows) Close() error {
 // should be taken when closing Rows not to modify
 // a buffer held in dest.
 func (r *Rows) Next(dest []driver.Value) error {
-	if r.currentRecord >= len(r.records) {
-		return io.EOF
-	}
-	record := r.records[r.currentRecord]
+	if r.currentRecord == nil || int64(r.currentRow) >= r.currentRecord.NumRows() {
+		if r.streamError != nil {
+			return r.streamError
+		}
```

Comment:
if we are going to have a mutex, it needs to be locked before these usages of `r.currentRecord`. But I still don't think we need a mutex for this. We can just pull the next record from the channel as we finish the current record. There wouldn't be any concurrent accessing of `r.currentRecord`

Reply:
deal. mutex removed.